File: TestApp/Tests/UI/ACTScreenshotter.m

#import <XCTest/XCTest.h>
#import "ACTScreenshotter.h"

@implementation ACTScreenshotter

+ (void)screenshotWithTitle:(NSString *)title {
    [XCTContext
     runActivityNamed:title
     block:^(id<XCTActivity>  _Nonnull activity) {
         XCUIScreenshot *screenshot = [[XCUIScreen mainScreen] screenshot];
         XCTAttachment *attachment;
         attachment = [XCTAttachment attachmentWithScreenshot:screenshot];
         [attachment setLifetime:XCTAttachmentLifetimeKeepAlways];
         [activity addAttachment:attachment];
     }];
}

@end

File: TestApp/Tests/UI/ACTScreenshotter.swift

import Foundation
import XCTest

class ACTScreenshotter {
    class func screenshotWithTitle (title: String) {
        XCTContext.runActivity(named: title) { (activity) in
            let screenshot = XCUIScreen.main.screenshot()
            let attachment = XCTAttachment(screenshot: screenshot)
            attachment.lifetime = .keepAlways
            activity.add(attachment)
        }
    }
}
